src/main/groovy/com/palantir/gradle/gitversion/JGitDescribe.groovy

```diff
@@ -0,0 +1,121 @@
+package com.palantir.gradle.gitversion
+
+import org.eclipse.jgit.api.Git
+import org.eclipse.jgit.internal.storage.file.FileRepository
+import org.eclipse.jgit.lib.Constants
+import org.eclipse.jgit.lib.ObjectId
+import org.eclipse.jgit.lib.Ref
+import org.eclipse.jgit.revwalk.RevCommit
+import org.eclipse.jgit.revwalk.RevWalk
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+/**
+ * JGit implementation of git describe with required flags. JGit support for describe is minimal and there is no support
+ * for --first-parent behavior.
+ */
+class JGitDescribe implements GitDescribe {
+    private static final Logger log = LoggerFactory.getLogger(JGitDescribe.class)
+
+    private File directory
+
+    JGitDescribe(File directory) {
+        this.directory = directory
+    }
+
+    @Override
+    String describe(String prefix) {
+        Git git = Git.wrap(new FileRepository(GitCli.getRootGitDir(directory)))
+        if (!GitUtils.isRepoEmpty(git)) {
+            log.debug("Repository is empty")
+            return null
+        }
+
+        RevCommit headCommit
+        try {
+            ObjectId headObjectId = git.getRepository().resolve(Constants.HEAD)
+            RevWalk walk = new RevWalk(git.getRepository())
```

Comment:
@iamdanfox there is some groovy weirdness happening here because walk is used in line 47

Reply:
Yeah I just noticed that, will revert my commit but I'm kinda horrified the tests are passing

Reply:
There is a huge bug. Tests aren't running correctly :/ Do not merge